### DISCUSSION TITLE
Add timeout option to Pilot/BuildWatcher

### DISCUSSION THREAD

**Discussion by sebastianludwig** [fastlane/fastlane#17814] [General] [answered]
Every now and then we have builds stuck in ASC in the "Waiting for processing..." phase. As described in #14997 I understand that this is not fastlane's fault. We'd still like to be able to specify a maximum time a build can block our CI nodes. As @joshdholtz suggested in https://github.com/fastlane/fastlane/issues/14952#issuecomment-506062825 a `timeout`/`max_wait_time` option would be great.

**Comment by rogerluan** [comment]
Hey @sebastianludwig 👋 

Your request makes total sense, I support it! 🙏 

Just wondering, when/why do your builds get "stuck"? How do you get them out of that state? 🤔

**Comment by sebastianludwig** [comment]
If I remember correctly it happened three times over the last two month. One was a false positive where fastlane was still waiting but the build was already distributed and available via TestFlight. The other two were in a sister project and I just killed the Jenkins job without further investigation after 9 hours 😬

**Comment by crazymanish** [comment, marked as answer]
FYI @sebastianludwig 
Open PR here: https://github.com/fastlane/fastlane/pull/18660

**Reply by rogerluan** [reply]
https://github.com/fastlane/fastlane/pull/18660 has been merged and release in https://github.com/fastlane/fastlane/releases/tag/2.184.0 🚀
